sycl/source/detail/platform_impl.cpp

```diff
@@ -121,95 +122,172 @@ vector_class<platform> platform_impl::get_platforms() {
 }
 
 struct DevDescT {
-  const char *devName = nullptr;
-  int devNameSize = 0;
-  const char *devDriverVer = nullptr;
-  int devDriverVerSize = 0;
-
-  const char *platformName = nullptr;
-  int platformNameSize = 0;
-
-  const char *platformVer = nullptr;
-  int platformVerSize = 0;
+  std::string devName;
+  std::string devDriverVer;
+  std::string platName;
+  std::string platVer;
 };
 
 static std::vector<DevDescT> getAllowListDesc() {
-  const char *str = SYCLConfig<SYCL_DEVICE_ALLOWLIST>::get();
-  if (!str)
+  std::string allowList(SYCLConfig<SYCL_DEVICE_ALLOWLIST>::get());
```

Comment:
Please use coding guide-lines: https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly

Reply:
Done.